### PR TITLE
adding TestSplitStorage.test_archival_node_sync

### PR DESCRIPTION
**Pull request by posvyatokum**
Adding localnet test to make sure that node can sync from cold storage.

Introducing _migrate_to_split_storage a.k.a moving code around. Using it in old migration test (literally nothing changed).
The main part of this PR is test_archival_node_sync -- spinning up legacy archival node, split storage archival node + validator and rpc node (to make split storage node split). 
Test flow is as follows:
1. Migrate split storage node
2. Kill legacy archival node
3. Produce bunch more epochs (twice as more as number of epochs to keep by split storage node)
4. Kill validator
5. Restart legacy archival node
6. Check that it is syncing a bit

Will restart Nayduck when it will feel better, but for now no red flags, one green flag https://nayduck.near.org/#/run/2934

### CODE REVIEW

File: pytest/tests/sanity/split_storage.py [hunk 1]
```diff
@@ -76,6 +76,126 @@ def _check_split_storage_info(
...
+        logger.info("")
+
+        self._configure_cold_storage(archival_dir)
+        archival.kill(gentle=True)
+        archival.start()
+
+        # Wait for a few seconds to:
+        # - Give the node enough time to get started.
+        # - Give the cold store loop enough time to run - it runs every 1s.
+        # TODO(wacban) this is a quick stop-gap solution to fix nayduck, this
```
Comment (nikurt) [inline pytest/tests/sanity/split_storage.py:108]:
```suggestion
        # TODO(posvyatokum) this is a quick stop-gap solution to fix nayduck, this
```

File: pytest/tests/sanity/split_storage.py [hunk 2]
```diff
@@ -76,6 +76,126 @@ def _check_split_storage_info(
         else:
             self.assertIsNone(cold_head_height)
 
+    def _migrate_to_split_storage(self, rpc, archival, archival_dir,
```
Comment (nikurt) [inline pytest/tests/sanity/split_storage.py:79]: Add a comment that gives a high-level explanation of what's this test does.
The PR has a good such description.

File: pytest/tests/sanity/split_storage.py [hunk 3]
```diff
@@ -76,6 +76,126 @@ def _check_split_storage_info(
...
+            expected_hot_db_kind="Archive",
+            # The cold storage head should be fully caught up by now.
+            check_cold_head=True,
+        )
+
+        logger.info("")
+        logger.info("Phase 3 - Preparing hot storage from rpc backup.")
+        logger.info("")
+
+        # TODO Ideally we won't need to stop the node while running prepare-hot.
```
Comment (nikurt) [inline pytest/tests/sanity/split_storage.py:135]: How can you not stop the node to do the following manipulations?
You're making a massive write to the database.

**Review by nikurt — APPROVED** [review]

File: pytest/tests/sanity/split_storage.py (hunk 3 above)
Comment (posvyatokum) [reply pytest/tests/sanity/split_storage.py]: prepare-hot command only changes the DBKind of provided rpc db. Archival node is needed to make sure that rpc db is compatible with it as future hot db.
So, actually archival node can keep running. And actually it already can (this code was written before some changes in cold store tool). Still, archival node will need a restart in the next phase, so the kill command just moves here.

**Pull request merged**